endpoints/s3/src/putobject.cpp

```diff
@@ -131,16 +154,32 @@ class incremental_async_read
             self->total_bytes_read_ += byte_count;
             logging::debug("{}: multipart upload: Total bytes [{}] read for part file [{}].", __func__, self->total_bytes_read_, self->part_filename_);
 
-            if (!self->part_file_.write(self->buffer_.data(), byte_count)) {
-                logging::error("{}: multipart upload: Error writing [{}] bytes to part file [{}].", __func__, byte_count, self->part_filename_);
-                self->resp_.result(beast::http::status::internal_server_error);
-                self->session_ptr_->send(std::move(self->resp_)); // Schedules an async write op.
-                return;
+            if (self->upload_part_flag_) {
+                if (!self->part_file_.write(self->buffer_.data(), byte_count)) {
+                    logging::error("{}: multipart upload: Error writing [{}] bytes to part file [{}].", __func__, byte_count, self->part_filename_);
+                    self->resp_.result(beast::http::status::internal_server_error);
+                    self->session_ptr_->send(std::move(self->resp_)); // Schedules an async write op.
+                    return;
+                }
+                logging::debug("{}: multipart upload: Wrote [{}] bytes to part file [{}].", __func__, byte_count, self->part_filename_);
+            } else {
+                if (!self->odstream_.write(self->buffer_.data(), byte_count)) {
+                    logging::error("{}: multipart upload: Error writing [{}] bytes iRODS file [{}].", __func__, byte_count, self->irods_path_);
```

Comment:
"bytes iRODS file" -> "bytes to iRODS data object"?

Reply:
done